backend/server.py

```
@@ -1517,6 +1528,20 @@ async def _deferred_startup_cleanup() -> None:
+        # Nuclear cleanup: bulk-delete duplicates directly in the DB. This is
+        # the fix for the 2100+ schedule multiplication bug — force_cleanup
+        # only deduplicates in-memory jobs, but if 2100 duplicates are
+        # persisted in the DB (from a previous bug), force_cleanup alone
+        # won't catch them all. nuclear_cleanup deduplicates by name.
+        try:
+            from packages.scheduler.cleanup import nuclear_cleanup
+            nuclear = await nuclear_cleanup(get_db())
+            if nuclear.get("deduped", 0) > 0 or nuclear.get("deleted_run_once", 0) > 0 or nuclear.get("deleted_stuck", 0) > 0:
+                log.info("Startup nuclear cleanup: run_once=%d stuck=%d deduped=%d total_remaining=%d",
+                         nuclear["deleted_run_once"], nuclear["deleted_stuck"], nuclear["deduped"], nuclear["total"])
+        except Exception as exc:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)